src/minion-ffi/tests/runner.c

```diff
@@ -177,15 +187,22 @@ int main(int argc, const char** argv) {
         close(comm_pipe[1]);
         close(devnullfd);
         const char* output;
-        size_t output_sz = read_all(comm_pipe[0], &output);
+        int is_timeout;
+        size_t output_sz = read_all(comm_pipe[0], &output, &is_timeout);
         const char* output0 = output;
         size_t expected_output_sz = strlen(tests[i].expected_output);
         if (output_sz != expected_output_sz ||
-            memcmp(output, tests[i].expected_output, output_sz)) {
+            memcmp(output, tests[i].expected_output, output_sz) ||
+            is_timeout) {
             fail = 1;
+            fprintf(stderr, "test `%s`: ", tests[i].name);
+            if(is_timeout)
+                fprintf(stderr, "timeout");
+            else
+                fprintf(stderr, "output differs");
             fprintf(stderr,
-                    "test `%s`: output differs:\nActual output (len %lld):\n",
-                    tests[i].name, (long long) output_sz);
+                    ":\nActual output (len %lld):\n",
+                    (long long) output_sz);
```

Comment:
Do you kill child?

Reply:
Very interesting question... probably no